sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationSettingTests.cs

```diff
@@ -34,13 +34,12 @@ public void FilterReservedCharacter()
                 Labels = new List<string>() { "my_label", "label,label" },
             };
 
-            var builder = new RequestUriBuilder
-            {
-                Uri = new Uri("http://localhost/")
-            };
+            var builder = new RequestUriBuilder();
+            builder.Assign(new Uri("http://localhost/"));
```

Comment:
Could the property be called BaseUri and remain get set property? I just did not like that the property was logically "the build method". A BaseUri property would be just like the others, i.e. a component of the final Uri.

Reply:
I liked how it matched `UriBuilder`.

Reply:
If we are not trying to match `UriBuilder` I want to make it clear that assigning `BaseUri` resets everything.

Reply:
Agreed that `BaseUri` feels dangerous.  I think taking it in the .ctor makes more sense than `Assign` and you can create a new instance if you want to reset the entire URI.

Reply:
I'd like to discourage people from creating instances or the builder, it can be to easily overused wasting a lot of resources.

Reply:
What's the difference in resource usage between creating a new builder vs. calling `Assign`?  Is the idea is `Assign` just makes it harder to use so people are less likely to?

Reply:
Assign reuses the internal StringBuilder and if it has enough space reuses the internal `char[]`. Creating new builder allocates uri builder itself, string builder and the buffer.

Reply:
Got it.  I don't really like how you have to construct then assign, but I don't think a ton of people are going to be exposed to this type.  I'm more concerned with Storage UriBuilder types that we do expect a decent number of customers to interact with.  _(I'm assuming everyone agrees we should match Azure.Core's UriBuilder and add `Assign`/`ToUri`.)_

Reply:
I don't think matching Azure.Core's UriBuilder every method is not necessary. Having a ctor(Uri) in storage UriBuilder type is a right thing for customers.

Reply:
I'm fine with that.  @KrzysztofCwalina - are you okay with `BlobUriBuilder` having a less efficient but slightly more approachable API that differs from `RequestUriBuilder` in Azure.Core?

```
    RequestUriBuilder rb = new RequestUriBuilder();
    rb.Assign(new Uri("..."));
    // ...
    Console.WriteLine(rb.ToUri());
     
    
    BlobUriBuilder bb = new BlobUriBuilder(new Uri("..."));
    // ...
    Console.WriteLine(bb.Uri);
```

Reply:
@tg-msft why not keep assign and ToUri just add a ctor to make it more user friendly?

Reply:
Per the discussion today, you'll rename `Assign` to `Reset` and we'll keep our .ctor but add `ToUri`.